### PR TITLE
feat(FOROME-811): add disabling applying button in setting filter

### PR DESCRIPTION
**Pull request by QSdmitrioul**
https://quantori.atlassian.net/browse/FOROME-811

### CODE REVIEW

**Review by QYaroslavTrefilov — COMMENTED** [review]
When entering step edit mode, the button turns gray only after changing the step. It seems to me that the button should turn gray immediately after switching to step change mode. Maybe we should ask Irina Grechkina?

**Review by QYaroslavTrefilov — APPROVED** [review]

File: src/pages/filter/ui/selected-group.tsx [hunk 1]
```diff
@@ -17,6 +17,17 @@ export const SelectedGroup = observer((): ReactElement => {
 
   const { isRedactorMode } = filterStore
 
+  const getGroup = () => {
```
Comment (AlMaQntr) [inline src/pages/filter/ui/selected-group.tsx:20]: Please don't use closure function here. You can use simple assignment with `group` variable.
Or if you want, you may create independent component e.g. <SelectedGroupPanel />

**Comment by QSdmitrioul** [comment]
> 



> When entering step edit mode, the button turns gray only after changing the step. It seems to me that the button should turn gray immediately after switching to step change mode. Maybe we should ask Irina Grechkina?

It is her idea, it become disabled only if we have unsaved changes

**Comment by AlMaQntr** [comment]
Overall, i think it's not a good way to check that someone changed in edit mode. Because you put `setIsChanged(...)` and `setIsNotChanged(...)` multiple times in event handlers in your code. 
May be we can create this behaviour more centralised with comparing initialValue of filter and it's current value in edit panel?

Need discuss

**Review by AlMaQntr — COMMENTED** [review]
Overall, i think it's not a good way to check that someone changed in edit mode. Because you put `setIsChanged(...)` and `setIsNotChanged(...)` multiple times in event handlers in your code. 
May be we can create this behaviour more centralised with comparing initialValue of filter and current value in edit panel?

Need discuss

**Review by Lapk1n — APPROVED** [review]

**Pull request closed without merging**